azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -137,6 +138,30 @@ public void uploadExecutableFlow(final ExecutableFlow flow)
     }
   }
 
+  public List<ExecutableFlow> fetchStaleFlows(final long beforeInMs)
+      throws ExecutorManagerException {
+    final List<Status> finalStates = ImmutableList
+        .of(Status.KILLED, Status.FAILED, Status.SUCCEEDED);
+    final StringBuilder query = new StringBuilder(FetchExecutableFlows.FETCH_FLOWS_STARTED_BEFORE);
+    query.append(" AND status NOT IN (");
+    query.append(
+        finalStates.stream()
+            .map(s -> String.valueOf(s.getNumVal()))
+            .collect(Collectors.joining(", ")));
+    query.append(")");
+
```

Comment:
Can you add a comment stating how the query will look in final form?

Reply:
Will do.